daffodil-core/src/main/scala/org/apache/daffodil/compiler/Compiler.scala

```diff
@@ -302,7 +302,7 @@ class Compiler(var validateDFDLSchemas: Boolean = true)
    * to see if compilation was successful or not.
    */
   def compileFile(file: File): ProcessorFactory = {
-    val source = URISchemaSource(file.toURI)
+    val source = new URISchemaSource(file.toURI)
```

Comment:
All these 'new' insertions could be reverted if an object with an apply method was created. That would really reduce the clutter. If you make the constructor protected, then these will all pop up in the compiler as illegal and can be fixed in bulk.

Reply:
Fixed